### PR TITLE
Enable `aten.t.default` conversion for more cases

### PR DESCRIPTION
**Pull request by jerrysky3**
### Ticket
#377

### Problem description
Enable `aten.t.default` conversion for more cases and fallback for the issue mentioned in #377

### What's changed
- Explicitly handle unsupported cases in `to_tt_pass.py`
- Remove `aten.t.default` from blocklist
- Add tests to verify support and unsupported cases

### CODE REVIEW

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 1]
```diff
@@ -634,16 +634,27 @@ def rewrite_node(node):
                     return g.call_function(ttnn.reshape, args=(args[0], output_size))
                 return None
 
-            if node.target == torch.ops.aten.transpose.int:
-                dim0 = args[1]
-                dim1 = args[2]
+            if node.target in [torch.ops.aten.transpose.int, torch.ops.aten.t.default]:
                 output_size = node.meta["val"].size()
                 rank = len(output_size)
+                if node.target == torch.ops.aten.t.default:
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/to_tt_pass.py:640]: what's the difference between `torch.ops.aten.transpose` and `ops.aten.t`?

Comment (jerrysky3) [reply torch_ttnn/passes/lowering/to_tt_pass.py:640]: I guess there are historical reasons but `aten.t` only transposes 2D tensor and `aten.transpose` can swap any two dims on n-D tensor

https://pytorch.org/docs/stable/generated/torch.t.html
https://pytorch.org/docs/stable/generated/torch.transpose.html

File: torch_ttnn/passes/lowering/to_tt_pass.py [hunk 2]
```diff
@@ -634,16 +634,27 @@ def rewrite_node(node):
...
+                    assert rank >= 0 and rank <= 2, "Input tensor can only be 0D, 1D or 2D"
```
Comment (ayerofieiev-tt) [inline torch_ttnn/passes/lowering/to_tt_pass.py:641]: instead of asserting, should we fallback?

Comment (jerrysky3) [reply torch_ttnn/passes/lowering/to_tt_pass.py:641]: Given the pytorch spec, this shouldn't be possible: https://pytorch.org/docs/stable/generated/torch.t.html

"Expects input to be <= 2-D tensor and transposes dimensions 0 and 1."

So I think it makes sense to use assert here

Comment (ayerofieiev-tt) [reply torch_ttnn/passes/lowering/to_tt_pass.py:641]: Thank you!

**Review by ayerofieiev-tt — APPROVED** [review]

File: torch_ttnn/passes/lowering/to_tt_pass.py (hunk 1 above)
Comment (ayerofieiev-tt) [reply torch_ttnn/passes/lowering/to_tt_pass.py:640]: noted, thank you

**Pull request merged**